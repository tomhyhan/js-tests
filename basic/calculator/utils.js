function add(x,y) {
    return x+y
}

function sub(x,y) {
    return x-y
}

function mul(x,y) {
    return x*y
}

function div(x,y) {
    if (y == 0) {
        throw new Error("Can not divide by 0")
    }
    return x / y
}

module.exports = {add, sub, div, mul}