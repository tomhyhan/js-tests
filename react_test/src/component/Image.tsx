import React from 'react'
import { ImageT } from '../lib/Types/ImageT';
import { useNavigate } from 'react-router-dom';

export default function Image({image}: {
    image: ImageT
}) {
    const navigate = useNavigate()

    const handleImageClick = () => {
        navigate(`/images/${image.id}`, {
            state: { image },
        })
    }

    return (
        <li className="px-4 cursor-pointer" onClick={handleImageClick}>
            <img src={image.download_url} alt={image.author} onClick={handleImageClick}/>
        </li>
    )
}
