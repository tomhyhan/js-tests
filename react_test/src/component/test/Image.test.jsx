import React from "react"
import { describe, it } from 'vitest'
import { withRouter } from "../../test/utils"
import {Route} from "react-router-dom"
import Image from "../Image"
import { fakeImage } from "../../test/fakeData"
import {render, screen} from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import '@testing-library/jest-dom';

describe("Image",()=> {
    it("renders Image",() => {
        const { asFragment } = render(withRouter(
            <Route path="/" element={<Image image={fakeImage}/>}/>
        ))

        expect(asFragment()).toMatchSnapshot()
    })

    it("navigate to image detail page", async () => {
        render(withRouter(
            <>
                <Route path="/" element={<Image image={fakeImage}/>}/>
                <Route path={`/images/${fakeImage.id}`} element={imageDetailComponent()}/>
            </>
        ))

        const link = screen.getByRole("img")
        await userEvent.click(link)
        expect(screen.getByText(JSON.stringify(fakeImage))).toBeInTheDocument()
    })

    function imageDetailComponent() {
        return <pre>{JSON.stringify(fakeImage)}</pre>
    }
})

