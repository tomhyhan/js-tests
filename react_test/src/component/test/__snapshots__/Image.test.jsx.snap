// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Image > renders Image 1`] = `
<DocumentFragment>
  <li
    class="px-4 cursor-pointer"
  >
    <img
      alt="Bar"
      src="https://foo.bar.download_url/"
    />
  </li>
</DocumentFragment>
`;
